packages/next/src/shared/lib/router/utils/route-regex.ts

```diff
@@ -99,48 +100,77 @@ function buildGetSafeRouteKey() {
   }
 }
 
+function getSafeKeyFromSegment({
+  segment,
+  routeKeys,
+  keyPrefix,
+}: {
+  segment: string
+  routeKeys: Record<string, string>
+  keyPrefix?: string
+}) {
+  const getSafeRouteKey = buildGetSafeRouteKey()
+
+  const { key, optional, repeat } = parseParameter(segment)
+
+  // replace any non-word characters since they can break
+  // the named regex
+  let cleanedKey = key.replace(/\W/g, '')
+
+  if (keyPrefix) {
+    cleanedKey = `${keyPrefix}${cleanedKey}`
+  }
+  let invalidKey = false
+
+  // check if the key is still invalid and fallback to using a known
+  // safe key
+  if (cleanedKey.length === 0 || cleanedKey.length > 30) {
+    invalidKey = true
+  }
+  if (!isNaN(parseInt(cleanedKey.slice(0, 1)))) {
+    invalidKey = true
+  }
+
+  if (invalidKey) {
+    cleanedKey = getSafeRouteKey()
+  }
+
+  if (keyPrefix) {
+    routeKeys[cleanedKey] = `${keyPrefix}${key}`
+  } else {
+    routeKeys[cleanedKey] = `${key}`
+  }
+
+  return repeat
+    ? optional
+      ? `(?:/(?<${cleanedKey}>.+?))?`
+      : `/(?<${cleanedKey}>.+?)`
+    : `/(?<${cleanedKey}>[^/]+?)`
+}
+
 function getNamedParametrizedRoute(route: string, prefixRouteKeys: boolean) {
   const segments = removeTrailingSlash(route).slice(1).split('/')
-  const getSafeRouteKey = buildGetSafeRouteKey()
   const routeKeys: { [named: string]: string } = {}
   return {
     namedParameterizedRoute: segments
       .map((segment) => {
-        if (segment.startsWith('[') && segment.endsWith(']')) {
-          const { key, optional, repeat } = parseParameter(segment.slice(1, -1))
-          // replace any non-word characters since they can break
-          // the named regex
-          let cleanedKey = key.replace(/\W/g, '')
-
-          if (prefixRouteKeys) {
-            cleanedKey = `${NEXT_QUERY_PARAM_PREFIX}${cleanedKey}`
-          }
-          let invalidKey = false
-
-          // check if the key is still invalid and fallback to using a known
-          // safe key
-          if (cleanedKey.length === 0 || cleanedKey.length > 30) {
-            invalidKey = true
-          }
-          if (!isNaN(parseInt(cleanedKey.slice(0, 1)))) {
-            invalidKey = true
-          }
-
-          if (invalidKey) {
-            cleanedKey = getSafeRouteKey()
-          }
-
-          if (prefixRouteKeys) {
-            routeKeys[cleanedKey] = `${NEXT_QUERY_PARAM_PREFIX}${key}`
-          } else {
-            routeKeys[cleanedKey] = `${key}`
-          }
-
-          return repeat
-            ? optional
-              ? `(?:/(?<${cleanedKey}>.+?))?`
-              : `/(?<${cleanedKey}>.+?)`
-            : `/(?<${cleanedKey}>[^/]+?)`
+        const markerMatches = segment.match(/\((\.{1,3})\)(\w*)/) // Check for intercept markers
```

Comment:
for the sake of maintainability, for the intercept marker at least, can you use the same method/variables used here https://github.com/vercel/next.js/blob/ea8c4274b818be0cebad1fa1b9ebc3202ee5592c/packages/next/src/lib/generate-interception-routes-rewrites.ts#L25 ? just so it's easier to see where we check for interception routes

this also makes it easier for us to add more/less interception markers if need be